kitsune/sumo/utils.py

```diff
@@ -128,6 +128,10 @@ def get_next_url(request):
     else:
         url = request.META.get("HTTP_REFERER")
 
+    # Remove any newline or carriage-return characters.
+    if url:
+        url = url.replace("\n", "").replace("\r", "")
+
     if not settings.DEBUG and not is_safe_url(
```

Comment:
`is_safe_url()` has been renamed to `url_has_allowed_host_and_scheme()`, and the old name is deprecated, so we could change this while we are here.